Roundstart Species Design Guidelines

FairlySadPanda opened the pull request:
# Species Guidelines

This document covers the policy for the acceptance and maintenance of proposed playable roundstart species on Wizard's Den servers. This document exists as both the direct rules and a further set of soft principles for proposing, developing and maintaining roundstart species for Space Station 14. 

---

I've written this as I've threatened to a few times on the Discord and finally decided to do so this evening.

There's probably more than one alternative proposal in the works for this; this is ergo something of a focus point of discussion as well as a proposal in its own rights.

I've drafted these rules and guidelines after a lot of discussion on the Wizden Discord, on and off, pretty much since the start of 2024. They encapsulate my best effort at covering the views I encountered in these discussions, which I have then synthesized and, inevitably, coloured with my own opinion and views.

I'm happy to iterate on this document in response to feedback.

FairlySadPanda commented:
"Panda why are Plasmamen specifically required to suffer in your proposal"
"Because they're flaming skeletons and are automatically the coolest thing in the universe otherwise"

ScarKy0 commented:
holy shit

slarticodefast commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,124 @@
+
+Remember that the most invested person in a character or creature design is always the creator. It is always on the proposer of a species to prove why a species is sufficiently interesting to be a candidate for upstream.
+
+### Rule 2: New Features and Tech Improvements
+
+A species **must** represent an opportunity for evolving the Space Station 14 codebase. This can involve either a completely new feature or a significant remix of an existing feature, representing the opportunity for code improvement.
+
+It is not uncommon for a species to be proposed that is almost or completely achievable via YAML and assets alone. These species are often ideal for downstream servers, but would not meaningfully improve the ability for Wizard's Den to iterate on the game and make it better for everyone. This is more important than supporting every niche within the wider Space Station 14 playerbase (see The Golde ...
+
+In certain circumstances, a species that demonstrates a sufficiently interesting use of current mechanics, potentially in service of some other in-development content, **may** be granted exemption from this rule (see The Golden Rule above).
```
I think in general I agree with the current formulation.
It does not say anything about having to introduce a new mechanic, but that it needs to be *interesting* mechanically.
I would remove the "achievable via YAML" part because that is a bad way to measure mechanical complexity. A few lines of yaml can do a lot, and it might be that the component you need has already been written.

slarticodefast reviewed (commented):
Thanks a lot for writing this, I really like the approach you have taken. Here are some excerpts from my current notes on my attempt on a species doc, maybe you can incorporate some of them if you think they fit:


- Definition: What is a species?
  Defined in a SpeciesPrototype, which contains information about possible sprite layers, markings, name generation scheme, default entity prototype to spawn with etc.
  Species: Human, Lizard, Diona
  Not a species: Kobolts, Monkeys, Diona Nymphs, Corgis, Ninjas, Revenants (these are all custom mobs and some can come in random colors, but don't have a species prototype), Paradox Clones (these are a copy of an already spawned humanoid and have the same species as the original, but are not a species themselves).
- Roundstart species: A species that can be selected in the lobby's character editor and allows players to spawn at the start of a round in a normal job. This document focuses on the proposal of roundstart species. Non-roundstart species will have to follow our general [proposal template](https://github.com/space-wizards/docs/blob/master/src/en/templates/proposal.md) as well, especially for ghost roles. Examples:
  Roundstart species: Human, Lizard, Diona
  Not a roundstart species: Skeleton, Gingerbread man (admeme only), terminator (removed)
- Lore: While we have no official lore (at the moment, some maintainers had plans for this) a new species should fit thematically into the space station universe. Specific background like "this species comes from Planet XYZ" is not required. But there is a certain, hard to grasp sci-fi theme and with corresponding tropes the game follows. Maybe someone can formulate this better than me :)
- Iconic Design: A good test for a unique visual identity is to check if the species can be identified from its silhouette (for example moth wings, dwarf heigth, lizard tail)
- Why restrict species this much in the first place? Why have such high requirements for a design doc?
  - Once added it is very unlikely that a species will ever be removed again (while technically possible with database migrations this would still be a major breaking change and remove player's characters), and adding a species upstream makes it available by default on most downstreams. Species have a major impact on the lore and design of SS14 as a whole, so they should be up to high quality standards.
  - TODO: Is there a number of species we are aiming for? Something like 20 would be way too high in my opinion and would become bloat, so this is a reason to restrict new species to interesting ones. If a species adds new mechanics, then it would automatically block other species from using those due to overlap. 

I would also add a list of technical requirements and some tips on how to implement them. A species needs:
- In case of a unique body shape it needs displacement maps for clothing and in-hand sprites so we don't have to redraw every single one of them. While this is possible (a lot of vox clothing has custom sprites) this serves as a fallback.
- Sprites for body parts/organs (unless it's re-using existing ones)
- A pattern or dataset for names and a way to automatically generate them. Coming up with a custom naming scheme can help to give your species more flavor. Look at NamingSystem.cs to see how the patterns work. Examples:
  - Human, Slime, Dwarf: `<first name> <last name>`
  - Diona: `The <object in nature> of <concept>`
  - Lizard: `<verb>s the <noun>`
  - Vox: `First name only`

- The parts put into the pattern are picked from a localized dataset of ftl strings for each species. This way they can contain thematically different phrases for the patterns. For some species they are different for males and females. Examples:
  - Moths: First names are based on latin names for different moth species, last names on myths or legends.
  - Vox: First names are made of random syllables, often with repeating patterns
  - Skeleton: First name replaced by the name of a bone.
- Audio for species-specific emotes. Make sure that they are open source, have a compatible licence and are converted to mono (so they work with positional audio).
- Markings for player customization. TODO: art guidelines. These should keep the species visual design consistent and not allow any extreme results unless that is part of the species' identity. Recent example: Colorful Vox.

I would also maybe add a list of good and bad examples for species traits. These should serve as reference, not as a checklist in a "I want my species to be accepted, so I'm picking one of each so it fulfills the requirements" way. It is recommended to design a species from the ground up around an interesting mechanic or design rather than adding one of these to a species because you have to. Not all of these are always required, exceptions can be made and the approval of docs will be decided on a case by case basis. Be creative, but not lazy!
- Good: Unique metabolism or body code mechanics, for example vox requiring nitrogen to breath and taking poison damage from oxygen.
- Bad: The same, but different (a mechanic that is already present on another species, but is only slightly changed). For example a species that breathes CO2 but takes poison damage from nitrogen would not be mechanically distinct enough from vox to warrant it being its own species. However, this would be fine in combination with other new mechanics.
- Okay: Damage modifiers as long as the don't provide a significant combat advantage.
- Good: Unique food preferences. Examples: Lizards only eat meat, moths eat cotton. Again this adds flavour to a species, but is not enough on its own to make it distinct.
- Bad: Movement speed increase. These have been declined in previous docs, as they provide a significant combat advantage.
- Bad: Features that are bad for accessibility reasons. Example from a previously declined doc: A species that has the "Flash" status effect be white instead of black to similate greater susceptibility to light. This would cause problems for players with photosensitivity issues. Another example would be forced colorblindness for a species.
- TODO: more examples

ScarKy0 commented:
A discussion about this doc has been opened at https://forum.spacestation14.com/t/roundstart-species-design-guidelines-477/22075

FairlySadPanda commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,124 @@
+5. A species must not be "deliberately" bad or unusually challenging to play (except for Plasmamen).
+6. A species must not represent a new challenge to moderation and adminning.
+7. A species must be technically feasible to implement and maintain.
+8. A species must not circumvent or be fatally held back from greatness by a common gameplay element of Space Station 14.
```
"A species must not entirely circumvent one or more common gameplay elements of Space Station 14."
"A species must not be designed with a specific, critical flaw that, without that flaw, would make the species overpowered compared to other species".

Think it needs to be two rules.

Circumvention of a mechanic is an IPC ignoring any sort of hunger/thirst-like mechanic. Leads to lopsided design that is hard to keep in line and fun.

Overpowered-but-flawed is roundstart skeletons. They are fast, immune to spacing and whatnot, but they can only heal with milk. Huge limitation which works for a free agent but would have constant issues as crew.

RedBookcase replied:
Probably a good idea to split it into two yea. Maybe compress it down to "A species must not be designed with a specific, critical flaw that would make the species overpowered if removed".

FairlySadPanda commented:
I have updated this document with feedback.

I have also updated this document with policy on review and removal of species.

FairlySadPanda commented on src/en/space-station-14/character-species/guidelines.md:
This is the only time I'm straying onto actual ways-of-working policy changes and it's a deliberate choice. This allows maintainers to safely front "maybe we should remove dwarfs" and allow for a way to handle this without a giant public argument.

ScarKy0 reviewed (approved):
I have no major issues with the doc

NoElkaTheGod commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,174 @@
+
+ ### Rule 4: Speciesism
+
+ Speciesism is banned on Wizard's Den in its entirety. As such, species **must not** directly encourage the "othering" of those that play those species - for example, causing health problems in those around them, or being incapable of socializing or interacting healthily with the rest of the crew.
```
I think this should be expanded to "no encouraging breaking Wizden rules" instead of just "no encouraging speciesism'. Sure, speciesism is very bad, but I don't think that a species that promotes ERP would be accepted either.

Simyon264 commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,174 @@
+ For example: a species suffers from oxygen poisoning. This means they must wear internals at all times, except in specific circumstances where oxygen is not present. This species **should not** punish the player by dying rapidly if their internals fail or are removed, and, if they do die, they **should** be revivable.
+
+ If a species is significantly challenging to play as, compared to normal round-start species, the following rules apply:
+ - It **must not** be able to be selected round-start by new players.
```
I would say this can be allowed, but have guidebook open for that species when a new player selects it.

FairlySadPanda replied:
![Uploading 7i6e3v-2589314892.jpg…]()

FairlySadPanda commented:
> You use the phrase "maintenance team", please use "maintainers" or "maintainer team" instead.

I'll maintain you in a minute 🔧

slarticodefast reviewed (changes requested):
One last point I would like to see added to the rules is the topic of accesibility.
A species should not be designed in a way that makes it non-accessible to some players.
Some examples that came up before were
- A brighter flash effect
- Forced colorblindness (which should be a general trait instead of species specific)

EmoGarbage404 reviewed (changes requested):
Detailed critique left in forums

https://forum.spacestation14.com/t/roundstart-species-design-guidelines-477/22075/14

VasilisThePikachu reviewed (changes requested):
I would also look at some other open reviews here, as I agree with most of them

FairlySadPanda commented:
Can't actually post on the thread so just responding to Slam:

```I think it kind of has to be vague. Any feature is going to have some sort of benefit for one job or another, and there will always be strictly speaking a “best” species for any given job. But what we don’t want is a “meta” species, such that any specific species is either locked out of or trivializes a noticeable part of a role. And where that line is drawn is kinda fuzzy and depends on the job/mechanic.

I’m not saying it couldn’t be written better, but I think it’s gonna be hard to not have it be sort of a greyzone.```

What is clear is that people _are_ OK with a species having upsides or alterations that help them in certain jobs, but are _not_ OK with design that is built around making that species good at a certain job.

Remember that the theme of this document is **design**, not **mechanics**. The important thing is the species' design. A species' **design** cannot feature being good at a specific job. A mechanic added to support the design might emergently help them with a job. But the mechanic will change over time as the game changes.

The distinction is:

"Dwarfs when examining objects can tell how tough they are roughly how much damage they've taken" is a mechanic that would be useful to a Dwarf engineer or some scientist-y jobs.

"Dwarfs mine faster because they're good at mining" is linking a species' concept to what job on the station they're good at.

FairlySadPanda commented:
# Modifications made

* Rebuilt modification and removal section. This section now covers good practice when maintaining a species, not formal policy. (@SlamBamActionman, others)
* Reworded the Golden Rule (@VasilisThePikachu)
* Added an example of improving a design concept to Rule 2 (@VasilisThePikachu)
* Removed reference to "humans, slimepeople" from Rule 3 (@VasilisThePikachu)
* Revised Rule 4 to cover general rule-breaking. (@NoElkaTheGod)
* Revised Rule 4 to better define how speciesism might pop up (@EmoGarbage404)
* Revised Rule 5 to be less restrictive, and articulate better the problem a challenging species could have on a team antagonist job like Nukies, and that it should be considered when designing the species. (@VasilisThePikachu)

## Modifications made specifically to cover all of @EmoGarbage404 's points

* Modified Rule 2 to strip out the codebase requirements and focussed it on design specifically.
* Broke out Rule 3 into three sections, and tried my best to get the job impact thing written more clearly. This is tricky to get right.
* Broke out Rule 8 to cover Emo's suggestions. Think it came out much better.

EmoGarbage404 reviewed (approved):
my issues with the proposal have been addressed sufficiently

ScarKy0 commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,238 @@
+
+Wizard's Den maintainers always have the final say on the content that they do or do not want on upstream. Everything in this document can be overruled by maintainer consent.
+
+If Wizard's Den declines to accept a new species, this should not be considered a snub of the species concept as a whole. The objective of Wizard's Den is to provide a 'vanilla' codebase for Space Station 14 servers run by many other groups, as well as providing a fun experience for players on the Wizard's Den servers specifically. Because of Wizard's Den's role as an upstream, it can't usually a ...
```
```suggestion
If Wizard's Den declines to accept a new species, this should not be considered a snub of the species concept as a whole. The objective of Wizard's Den is to provide a 'vanilla' codebase for Space Station 14 servers run by many other groups, as well as providing a fun experience for players on the Wizard's Den servers specifically. Because of Wizard's Den's role as an upstream, it can't usually accept content as easily as many downstreams can. Likewise, because Wizard's Den is considered a 'vanilla' variant of Space Station 14, there are many great potential concepts for a species that are too unusual or specific to one way Space Station 14 is played (such as on high role-play servers, highly chaotic or comedic servers, and so on) to be suitable for Wizard's Den specifically.
```

ScarKy0 commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,238 @@
+1. A species must have a **distinct identity** that is notably different in theme from the other species available on Wizard's Den.
+2. A species must represent **new or substantially remixed design ideas**.
+3. A species must be, at least conceptually, reasonably balanced with the other species available on Wizard's Den.
+4. A species design must not directly encourage speciesism or other rule-breaking will be rejected.
```
```suggestion
4. A species design that directly encourages speciesism or other rule-breaking will be rejected.
```
Weird wording.

ScarKy0 commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,238 @@
+
+Species **must** not be designed in such a way that a new player, picking a random character and rolling that species, would have an excessively negative experience due to lack of game knowledge. This **does not** mean a species cannot have a notable, potentially deadly, downside. However, that downside **should** be recoverable from, presented obviously to a player who has no prior context of th ...
+
+At a minimum, when picking a species which could be challenging to a new player, a guidebook entry exists to help that player play the species, and that guidebook entry **must** appear when that species is picked for the first time, and this page **must** be focussed on assisting the player with how to play the species, not simply being relegated to an encyclopedia entry on that species's mechani ...
```
```suggestion
At a minimum, when picking a species which could be challenging to a new player, a guidebook entry exists to help that player play the species, and that guidebook entry **must** appear when that species is picked for the first time, and this page **must** be focused on assisting the player with how to play the species, not simply being relegated to an encyclopedia entry on that species's mechanics.
```

ScarKy0 reviewed (changes requested):
small stuff

VasilisThePikachu commented on src/en/space-station-14/character-species/guidelines.md:
```suggestion
In general, maintaining a place for interested players to talk about the changes, where players can keep track of what is happening with the changes, and where feedback can be sought, is a good idea. This is a good use of a forum thread on our [developmet design discussion section](https://forum.spacestation14.com/c/development/design-discussions/65). Discord threads are not recommended as they may get sidetracked easily or be hard to read back afterwards, but they can work.
```

VasilisThePikachu reviewed (approved):
Sorry for the long review, I am currently on vacation and just found time.

FairlySadPanda commented on src/en/space-station-14/character-species/guidelines.md:
```diff
@@ -1,5 +1,238 @@
+If a species is significantly challenging to play as, compared to normal round-start species, the following rules apply:
+- It **must not** be able to be selected round-start by new players without clear warnings and tutorialization in place.
+- It **should not** be able to be picked if a random species is chosen for a midround ghost role. 
+- It **should** be given careful consideration on its impact on certain team antagonists, such as Nuclear Operatives, if that species is played as one of the team. (For example, a species that cannot wear shoes may be significantly disadvantaging to a Nuclear Operative team that can't provide that species with access to slip resistance othererwise.)
```
```suggestion
- It **should** be given careful consideration on its impact on certain team antagonists, such as Nuclear Operatives, if that species is played as one of the team. (For example, a species that cannot wear shoes may be significantly disadvantaging to a Nuclear Operative team that can't provide that species with access to slip resistance otherwise.)
```

FairlySadPanda commented:
more like here comes the vulps
